scripts/surrogate_eval.py

```diff
@@ -0,0 +1,365 @@
+
+    checkpoint_dir = SCRIPT_PATH / '..' / 'checkpoints'
+    transform_path = Path(os.environ.get('BUILDINGS_BENCH', '')) / 'metadata' / 'transforms'
+
+    #################### Model setup ####################
+
+    # remove "module" from state_dict keys to load weights without data parallel
+    # source: https://gist.github.com/IAmSuyogJadhav/bc388a871eda982ee0cf781b82227283
+    # from collections import OrderedDict
+    # def remove_data_parallel(old_state_dict):
+    #     new_state_dict = OrderedDict()
+
+    #     for k, v in old_state_dict.items():
+    #         name = k[7:] # remove `module.`
+    #         new_state_dict[name] = v
+    
+    #     return new_state_dict
+    if args.weather is not None and args.weather[0] == 'all':
+        args.weather = g_weather_features
+
+    model, loss, predict = model_factory(args.config, model_args)
+    model = model.to(local_rank)
+    model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[local_rank], output_device=local_rank)
+    model.load_state_dict(
+        torch.load(f"checkpoints/{args.checkpoint}", map_location=f'cuda:{local_rank}')["model"]
+    )
+    
+    print(f'rank {args.rank} number of trainable parameters is '\
+          f'= {sum(p.numel() for p in model.parameters())}', flush=True)
+
+    #################### Dataset setup ####################
+
+    building_dataset = load_pretraining('simcap',
+                            args.num_buildings,
+                            args.apply_scaler_transform,
+                            transform_path, 
+                            weather=args.weather, 
+                            context_len=0,
+                            pred_len=24,
+                            custom_idx_filename=args.test_idx_filename,
+                            use_buildings_chars=True,
+                            use_text_embedding=args.use_text_embedding
+                        )
+    
+    if not model.module.continuous_loads:
+        load_transform = LoadQuantizer(
+            with_merge=(not args.tokenizer_without_merge),
+            num_centroids=model.module.vocab_size,
+            device=f'cuda:{local_rank}')
+        load_transform.load(transform_path)
+    # else:
+    #     load_transform = train_dataset.load_transform
+    elif args.apply_scaler_transform != '':
+        load_transform = building_dataset.load_transform
+    else:
+        load_transform = None
+
+    if not model.module.continuous_loads: 
+        transform = load_transform.transform
+        inverse_transform = load_transform.undo_transform
+    elif args.apply_scaler_transform != '':
+        transform = lambda x: x
+        inverse_transform = load_transform.undo_transform
+    else: # Continuous unscaled values
+        transform = lambda x: x
+        inverse_transform = lambda x: x
+
+    model.eval()
+    step = 0
+
+    bd_types = ['FullServiceRestaurant', 'Hospital', 'LargeHotel', 'LargeOffice',
```

Comment:
This is also OK for now, but this is a good reminder that we need to make progress on writing code for the various qualitative evaluation metrics we will use